incubator/hnc/internal/validators/hierarchy_test.go

```diff
@@ -52,6 +52,49 @@ func TestStructure(t *testing.T) {
 	}
 }
 
+func TestChangeParentOnManagedBy(t *testing.T) {
+	f := foresttest.Create("-a-c") // a <- b; c <- d
+	h := &Hierarchy{Forest: f}
+	l := zap.Logger(false)
+
+	// Make c and d external namespaces
+	f.Get("c").ExternalTreeLabels = map[string]int{"c" + api.LabelTreeDepthSuffix: 0}
+	f.Get("d").ExternalTreeLabels = map[string]int{"d" + api.LabelTreeDepthSuffix: 0}
+
+	// The following 6 test cases matches the last 6 cases in the table at
```

Comment:
You should say which test does the other four. Also, maybe there's a better description than "the last 6 cases" - what makes those cases similar to each other so they can be tested together?